miniml/compiler/compile.scm

```diff
@@ -1245,7 +1260,9 @@
           ))
   (define (align shape args)
     (cond ((null? shape) (if (null? args) #nil (align (map (lambda (arg) (list 'Nolabel)) args) args)))
-          ((null? args) #nil)
+          ((null? args) (begin
+                          (for-each (lambda (lab) (assert (equal? (car lab) 'Nolabel))) shape)
```

Comment:
this `for-each` comes up also in `lower-var`; maybe factorize it as `(check-no-labels shape)`?